components/eamxx/src/physics/p3/impl/p3_update_prognostics_impl.hpp

```diff
@@ -78,9 +76,9 @@ ::update_prognostic_ice(
   qv.set(context, qv + (-qv2qi_vapdep_tend+qi2qv_sublim_tend-qv2qi_nucleat_tend)*dt);
 
   constexpr Scalar INV_CP = C::INV_CP;
-  th_atm.set(context, th_atm + inv_exner * ((qv2qi_vapdep_tend - qi2qv_sublim_tend + qv2qi_nucleat_tend) * (latvap+latice) * INV_CP +
-                                (qr2qi_collect_tend + qc2qi_collect_tend + qc2qi_hetero_freeze_tend + qr2qi_immers_freeze_tend -
-                                qi2qr_melt_tend + qc2qi_berg_tend) * latice * INV_CP) * dt);
```

Comment:
This usage of the constant `C::LatIce` is what is causing non-bfb to `latent_heat_fusion`. Investigating.

Reply:
Update: `LatIce` is exactly BFB with `latent_heat_fusion` for the entire run. Tested with:
```
  for (int p=0; p<Spack::n; ++p) {
    uint64_t latice_bits = *reinterpret_cast<const uint64_t*>(&latice);
    uint64_t fusion_bits = *reinterpret_cast<const uint64_t*>(&latent_heat_fusion[p]);
    if (latice_bits != fusion_bits)
      printf("latice:%.17f (%" PRIu64 "):"
            "latent_heat_fusion:%.17f (%" PRIu64 ")\n",
             latice,latice_bits,latent_heat_fusion[p],fusion_bits);
  }
```

Reply:
You said on weaver we are bfb right? If you haven't already, you could try to run a SMS_Ln1 CIME case, forcing mac_aero_mic::subcycling=1, save a bunch of output fields, and see which ones actually diff.